lib/core/macros.nim

```diff
@@ -1344,6 +1348,93 @@ proc copy*(node: NimNode): NimNode {.compileTime.} =
   ## An alias for copyNimTree().
   return node.copyNimTree()
 
+type GenAstOpt* = enum
+  kNoExposeLocalInjects,
+    # when unset, inject'd symbols (including implicit ones such as local procs
+    # in scope) are exposed implicitly;
+    # gensym'd symbols will generate a CT internal error: `environment misses:`
+    # when set, local symbols are not exposed unless captured explicitly in
+    # `genAst` argument list. The default is unset, to avoid surprising hijacking
+    # of local symbols by symbols in caller scope.
+
+macro genAst*(options: static set[GenAstOpt] = {}, args: varargs[untyped]): untyped =
```

Comment:
I am not a fan of this ``options`` argument at the beginning. AFAIK the default argument doesn't work here. So I always have to write the curly pair for genAst. Can't we get rid of these flags argument entirely, for example by providing two different names for genAst for the two different behaviors.

Reply:
I fixed it in latest commit (doens't appear here since you closed this PR)